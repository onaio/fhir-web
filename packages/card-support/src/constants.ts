// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
export const OPENSRP_URL_LOCATION_HIERARCHY = '/location/hierarchy';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
