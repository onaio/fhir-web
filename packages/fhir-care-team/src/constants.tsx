// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';
