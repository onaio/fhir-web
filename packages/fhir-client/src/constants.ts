export const dateFormat = 'yyyy-MM-dd';
