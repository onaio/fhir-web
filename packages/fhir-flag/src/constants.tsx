// urls
export const URL_ADMIN = '/admin';
export const URL_CLOSE_FLAGS = `${URL_ADMIN}/close-flags`;

// // magic strings

export const thatiMinutes = 30 * 60 * 1000;

// form fields
export const locationName = 'locationName' as const;
export const productName = 'productName' as const;
export const status = 'status' as const;
export const comments = 'comments' as const;
