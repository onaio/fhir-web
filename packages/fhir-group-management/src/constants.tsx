// url
export const ADD_EDIT_GROUP_URL = '/group/add-edit';
export const ADD_EDIT_COMMODITY_URL = '/commodity/add-edit';
export const LIST_GROUP_URL = '/groups/list';
export const LIST_COMMODITY_URL = '/commodity/list';

// magic strings
export const groupResourceType = 'Group';
export const listResourceType = 'List';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const type = 'type' as const;
export const unitOfMeasure = 'unitOfMeasure' as const;
