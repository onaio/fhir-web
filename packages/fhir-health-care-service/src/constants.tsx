// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTH_CARE_SERVICE_RESOURCE_TYPE = 'HealthcareService';
export const HEALTHCARES_ENDPOINT = 'HealthcareService/';
