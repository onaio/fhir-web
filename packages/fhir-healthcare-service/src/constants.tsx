// url
export const ADD_EDIT_HEALTHCARE_SERVICE_URL = '/healthcare/add-edit';
export const LIST_HEALTHCARE_URL = '/healthcare/list';

//
// constants
//
export const healthCareServiceResourceType = 'HealthcareService';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const comment = 'comment' as const;
export const extraDetails = 'extraDetails' as const;
export const providedBy = 'providedBy' as const;
