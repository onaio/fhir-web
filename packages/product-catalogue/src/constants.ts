// opensrp api strings
export const OPENSRP_API_BASE_URL = 'https://mg-eusm-staging.smartregister.org/opensrp/rest/';
export const OPENSRP_PRODUCT_CATALOGUE = 'product-catalogue';
export const PRODUCT_ID_ROUTE_PARAM = 'productId';

// router routes
export const CATALOGUE_LIST_VIEW_URL = '/product-catalogue/list';
export const CATALOGUE_EDIT_VIEW_URL = '/product-catalogue/edit';
export const CATALOGUE_CREATE_VIEW_URL = '/product-catalogue/new';
export const HOME_URL = '/';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'product-catalogue';

/** route params for product-catalogue pages */
export interface RouteParams {
  productId?: string;
}
