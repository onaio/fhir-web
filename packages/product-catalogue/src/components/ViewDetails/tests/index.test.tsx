import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Router } from 'react-router';
import { ViewDetails } from '..';
import { product1 } from '../../../ducks/productCatalogue/tests/fixtures';
import { createBrowserHistory } from 'history';
import { CATALOGUE_LIST_VIEW_URL } from '../../../constants';
import { act } from 'react-dom/test-utils';

const history = createBrowserHistory();

describe('View Details', () => {
  it('works correctly', () => {
    const props = { objectId: '1', object: product1 };
    const wrapper = mount(<ViewDetails {...props} />);
    expect(wrapper.text()).toMatchSnapshot('nominal display');
  });

  it('case without objectId', () => {
    const props = { objectId: '', object: product1 };
    const wrapper = mount(<ViewDetails {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot('Should be null');
  });

  it('works when objectId is but Object isnt', () => {
    const props = { objectId: '1', object: null };
    const wrapper = mount(
      <Router history={history}>
        <ViewDetails {...props} />
      </Router>
    );
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"404Sorry, the resource you requested for, does not existGo backGo home"`
    );
  });

  it('Closes on clicking cancle ', () => {
    const props = { objectId: '1', object: product1 };
    const wrapper = mount(
      <Router history={history}>
        <ViewDetails {...props} />
      </Router>
    );

    // simulate clicking on close button
    act(() => {
      wrapper.find('.flex-right button').simulate('click');
    });

    expect(wrapper.props().history.location.pathname).toEqual(CATALOGUE_LIST_VIEW_URL);
  });
});
