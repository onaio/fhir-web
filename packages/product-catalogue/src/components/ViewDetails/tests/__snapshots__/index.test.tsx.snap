// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Details case without objectId: Should be null 1`] = `
<ViewDetails
  extractViewDetails={[Function]}
  object={
    Object {
      "accountabilityPeriod": 1,
      "appropriateUsage": "staff is trained to use it appropriately",
      "availability": "available",
      "condition": "good condition",
      "isAttractiveItem": true,
      "materialNumber": "MT-123",
      "photoURL": "",
      "productName": "Scale",
      "serverVersion": 123456,
      "uniqueId": 1,
    }
  }
  objectId=""
/>
`;

exports[`View Details works correctly: nominal display 1`] = `"Product nameScaleUnique ID1Material numberMT-123Server version123456"`;
